examples/02-plot/sphere_eversion.py

```diff
@@ -0,0 +1,262 @@
+    x2 = x1 * exponential / denominator
+    y2 = y1 * exponential / denominator
+    z2 = numerator / denominator * exponential / gamma - (alpha - beta) / (alpha + beta) / gamma
+    return x2, y2, z2
+
+
+def unfold_sphere(theta, phi, t, q, eta, lamda):
+    """Unfold the sphere using Eqs. (12), (15), (10).
+
+    Input is a (theta, phi) point in spherical coordinates.
+    Output is an (x, y, z) point embedded in 3d space.
+
+    The parameter lamda varies. Lamda = 1 is the final stage of the
+    wormhole closing, and lamda = 0 is the recovered sphere.
+
+    """
+    # apply Eqs. (12), (15)
+    # fmt: off
+    x = (
+        t * (1 - lamda + lamda * np.cos(theta)**n) * np.cos(phi)
+        - lamda * w * np.sin(theta) * np.sin(phi)
+    )
+    x /= np.cos(theta)**n
+    y = (
+        t * (1 - lamda + lamda * np.cos(theta)**n) * np.sin(phi)
+        + lamda * w * np.sin(theta) * np.cos(phi)
+    )
+    y /= np.cos(theta) ** n
+    z = (
+        lamda * (
+            (w * np.sin(theta) * (np.sin(n * phi) - q * t)) / np.cos(theta)**n
+            - t / n * np.cos(n * phi)
+        )
+        - (1 - lamda) * eta**(1 + kappa) * t * abs(t)**(2 * kappa)
+            * np.sin(theta) / np.cos(theta)**(2 * n)  # noqa
+    )
+    # fmt: on
+
+    # apply Eq. (10)
+    denominator = x**2 + y**2
+    x2 = x * eta**kappa / denominator ** (1 - kappa)
+    y2 = y * eta**kappa / denominator ** (1 - kappa)
+    z2 = -z / eta / denominator
+    return x2, y2, z2
+
+
+###############################################################################
+# Now chain the functions by performing the process in Table 1 of the paper.
+# Start from the bottom for ``t = -1/Q``, keep stepping up, linearly changing
+# parameters that change from row to row, then at the top go from ``t = -1/Q``
+# to ``t = 1/Q``, then go back from top to bottom. Save each frame to a GIF.
+#
+# We make good use of the ``backface_params`` keyword parameter of
+# :func:`pyvista.Plotter.add_mesh`, allowing us to plot the inside and the
+# outside with different colors.
+
+# plot options to use for each frame
+opts = dict(
+    color='aquamarine',
+    specular=1.0,
+    specular_power=50.0,
+    backface_params=dict(color='forestgreen'),
+    smooth_shading=True,
+    reset_camera=True,
+)
+
+# use a small figure window to reduce the size of the GIF
+plotter = pv.Plotter(window_size=(300, 300))
+plotter.open_gif('sphere_eversion.gif')
+
+
+def save_frame(x, y, z):
+    """Helper to generate and store a frame of the eversion."""
+    plotter.clear()
+    plotter.enable_lightkit()
+    plotter.add_mesh(pv.StructuredGrid(x, y, z), **opts)
+    plotter.write_frame()
+
+
+# initial parameters, will be updated
+t = -1 / Q
+q = Q
+p = xi = alpha = 0
+eta = 1
+
+# sphere -> inverted wormhole
+theta, phi = np.mgrid[-np.pi / 2 : np.pi / 2 : 200j, -np.pi : np.pi : 400j]
+h, phi = sphere_to_cylinder(theta, phi)
+for lamda in np.linspace(0, 1, n_steps, endpoint=False):
```

Comment:
Codespell is complaining about my choice of `lamda` here. I want to follow the paper's convention to call this lambda, but of course `lambda` is a statement in Python so we can't use that. Is it OK to add `lamda` to codespell's ignore list? Not sure if that can lead to false negatives later.

Reply:
I'm fine with that.

Reply:
It would be ideal if codespell supported line/module/block level exclusion, but it doesn't.

https://github.com/codespell-project/codespell/issues/1212

One suggestion from this thread was to use --ignore-regex.  But "now you have two problems".  I'm also fine with simply adding it to the ignore list since codespell doesn't support this functionality directly.

Reply:
There is also `--exclude-file/-x` but it looks like it will be brittle to changes in style that may not be obvious to a contributor.

Reply:
Thanks for looking into that @MatthewFlamm. I don't think regex would work, because the word appears in all sorts of contexts in the file (otherwise I'd be happy with regex, I'm fairly fluent with it). So I think we're stuck with either ignoring the word or ignoring the file, and the former seems the less harmful option.

I've had some very busy weeks (I'm switching jobs pretty much as we speak) but I'll soon be able to follow up on this PR and others that I've been dragging my feet on.

Reply:
How about a name like `.codespellignore` that won't confuse contributors? I just added this at https://github.com/pyvista/pyvista-tutorial/pull/88.

Reply:
Thanks @banesullivan, I considered `lambda_` as an alternative, but since "lamda" is actually how many people write it (and how modern Greek people say and write it), I'm leaning towards keeping the current use and ignoring it for codespell.

Reply:
Added "lamda" to the codespell ignore list in https://github.com/pyvista/pyvista/pull/3135/commits/c1ff7101fa3369eae1d9a8d5b55e4caab9379065, thank you all for the inputs.